internal/backends/qt/qt_accessible.rs

```diff
@@ -9,51 +9,98 @@ use i_slint_core::item_tree::{ItemRc, ItemWeak};
-    accessible_item: *mut core::ffi::c_void,
+    accessible_item: *mut c_void,
 }
 
 impl i_slint_core::properties::PropertyChangeHandler for HasFocusPropertyTracker {
     fn notify(&self) {
         let accessible_item = self.accessible_item;
-        cpp!(unsafe [accessible_item as "Slint_accessible_item*"] {
-            QTimer::singleShot(0, [accessible_item]() {
-                // Delete this once we have returned from here: This is owned by the old
-                // QAccessibleInterface!
-                auto obj = accessible_item->object();
-                auto event = QAccessibleEvent(obj, QAccessible::Focus);
-                QAccessible::updateAccessibility(&event);
-            });
+        let data = cpp!(unsafe [accessible_item as "Slint_accessible_item*"] -> &SlintAccessibleItemData as "void*" {
+            auto obj = accessible_item->object();
+            auto event = QAccessibleEvent(obj, QAccessible::Focus);
+
+            QAccessible::updateAccessibility(&event);
 
-            accessible_item->armFocusTracker();
+            return accessible_item->data();
         });
+        data.arm_focus_tracker();
     }
 }
 
 pub struct AccessibleItemPropertiesTracker {
-    accessible_item: *mut core::ffi::c_void,
+    accessible_item: *mut c_void,
 }
 
 impl i_slint_core::properties::PropertyChangeHandler for AccessibleItemPropertiesTracker {
     fn notify(&self) {
         let accessible_item = self.accessible_item;
-        cpp!(unsafe [accessible_item as "Slint_accessible_item*"] {
-            QTimer::singleShot(0, [accessible_item]() {
-                // Delete this once we have returned from here: This is owned by the old
-                // QAccessibleInterface!
-                auto obj = accessible_item->object();
-                auto id = QAccessible::uniqueId(accessible_item);
-                auto new_ai = new Slint_accessible_item(accessible_item->takeRustItem(), obj,
-                    accessible_item->role(), accessible_item->parent());
-                QAccessible::deleteAccessibleInterface(id);
-                QAccessible::registerAccessibleInterface(new_ai);
-
-                auto event = QAccessibleStateChangeEvent(obj, new_ai->state());
-                QAccessible::updateAccessibility(&event);
-            });
+        let data = cpp!(unsafe [accessible_item as "Slint_accessible_item*"] -> &SlintAccessibleItemData as "void*"{
+            auto obj = accessible_item->object();
+
+            auto event = QAccessibleStateChangeEvent(obj, accessible_item->state());
+            QAccessible::updateAccessibility(&event);
+
+            return accessible_item->data();
+        });
+        data.arm_state_tracker();
+    }
+}
+
+pub struct SlintAccessibleItemData {
+    state_tracker:
+        Pin<Box<i_slint_core::properties::PropertyTracker<AccessibleItemPropertiesTracker>>>,
+    focus_tracker: Pin<Box<i_slint_core::properties::PropertyTracker<HasFocusPropertyTracker>>>,
```

Comment:
No need to use `Pin<Box>>` here
Just use #[pin_project]  and use `Pin<&SlintAccessibleItemData>`

(Can be done in a follow up)

Reply:
Follow up patch it is then:-)